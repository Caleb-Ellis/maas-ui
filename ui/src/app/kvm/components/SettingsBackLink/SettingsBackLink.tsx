import { Icon } from "@canonical/react-components";
import { Link, useHistory } from "react-router-dom";

type LocationState = {
  from?: string;
};

const SettingsBackLink = (): JSX.Element | null => {
  const history = useHistory();
  const state = history.location.state as LocationState;

  if (state === undefined) {
    return null;
  }

  let location = state.from as string;
  location = location.replace("/MAAS/r", "");

  return (
    <div className="settings-back-link">
      <Link className="settings-back-link__link" to={location}>
        <Icon className="u-rotate-right u-no-margin--left" name="chevron-up" />
        <span>Back</span>
      </Link>
      <hr className="settings-back-link__divider" />
    </div>
  );
};

export default SettingsBackLink;
