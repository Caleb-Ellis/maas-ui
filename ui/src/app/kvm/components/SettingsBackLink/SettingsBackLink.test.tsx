import { shallow } from "enzyme";
import { createMemoryHistory } from "history";
import { Router } from "react-router-dom";

import SettingsBackLink from "./SettingsBackLink";

describe("SettingsBackLink", () => {
  it("does not render is no from is provided", () => {
    let history = createMemoryHistory();
    const wrapper = shallow(
      <Router history={history}>
        <SettingsBackLink />
      </Router>
    );

    expect(wrapper.find("Link").exists()).toBe(false);
  });

  // XXX: Work out how to test this.
  xit("links to previous state when provided", () => {
    const expectedReturnURL = "/kvm/lxd/cluster/20/hosts";
    const locationState = `/MAAS/r${expectedReturnURL}`;
    const mockLocation = { state: { from: locationState } };
    const route = "/";
    const history = createMemoryHistory({
      initialEntries: [route, mockLocation.state.from],
    });
    const wrapper = shallow(
      <Router history={history}>
        <SettingsBackLink />
      </Router>
    );

    expect(wrapper.find("Link").prop("to")).toBe(`${expectedReturnURL}`);
  });
});
