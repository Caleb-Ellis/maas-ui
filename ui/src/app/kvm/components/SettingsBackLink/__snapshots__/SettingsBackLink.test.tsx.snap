// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SettingsBackLink renders 1`] = `
<Fragment>
  <div
    className="settings-back-link"
  >
    <Link
      className="settings-back-link__link"
      to="/kvm/lxd"
    >
      <Icon
        className="u-rotate-right"
        name="chevron-up"
      />
      <span>
        Back
      </span>
    </Link>
    <hr
      className="settings-back-link__divider"
    />
  </div>
</Fragment>
`;
